src/main/java/me/zeroeightsix/fiber/api/schema/type/EnumSerializableType.java

```diff
@@ -0,0 +1,49 @@
+package me.zeroeightsix.fiber.api.schema.type;
+
+import me.zeroeightsix.fiber.api.serialization.TypeSerializer;
+import me.zeroeightsix.fiber.impl.constraint.EnumConstraintChecker;
+
+import java.util.*;
+
+public final class EnumSerializableType extends SerializableType<String> {
+
```

Comment:
This can work with Java enums, no?
`class EnumSerializableType<E extends Enum<E>> extends SerializableType<E>`

Reply:
I don't like that very much as it means you get custom types in the intermediate representation. A `SerializableType` should be deserializable with no context from the application that created it.

Reply:
Fair enough. This way configs are completely defined by their schemas.